pandas/tests/indexes/base_class/test_replace.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
+
+def test_index_replace():
+    index = pd.Index([1, 2, 3])
+    expected = pd.Index(["a", 2, "c"])
+
+    result = index.replace([1, 3], ["a", "c"])
+
+    tm.assert_equal(result, expected)
+
+
+def test_index_replace_1():
+    index = pd.Index([1, 2, 3])
+    expected = pd.Index(["a", 2, 3])
+
+    result = index.replace(1, "a")
+
+    tm.assert_equal(result, expected)
+
+
+def test_index_replace_2():
+    index = pd.Index([1, 2, 3])
+    expected = pd.Index(["a", 2, "c"])
+
+    result = index.replace({1: "a", 3: "c"})
+    tm.assert_equal(result, expected)
+
+
+def test_index_replace_2_1():
+    index = pd.Index([1, 2, 3])
+
+    with pytest.raises(TypeError):
+        index.replace({1: "a", 3: "c"}, "x")
+
+
+def test_index_replace_3():
+    index = pd.Index([1, None, 2])
+    with pytest.raises(NotImplementedError):
```

Comment:
Can you add the ```match``` argument, and the expected error message

Reply:
Are we sure we want to raise a `NotImplementedError` here? I'd rather not receive one from a pandas. I'll wait until that's resolved and add the message accordingly.